src/Makefile

```diff
@@ -58,77 +58,82 @@ include $(top_srcdir)/config/config.mk
 # Source files
 
 # Add standalone C++ header files to this list
-CCHDR =  exiv2.hpp         \
-         config.h          \
-         exv_conf.h        \
-         xmp_exiv2.hpp     \
+CCHDR =  exiv2.hpp             \
+         config.h              \
+         exv_conf.h            \
+         xmp_exiv2.hpp         \
          version.hpp
 
 # Add library C++ source files to this list
-CCSRC =  basicio.cpp       \
-	 bmpimage.cpp          \
-	 bigtiffimage.cpp      \
-	 canonmn.cpp           \
-	 casiomn.cpp           \
-	 convert.cpp           \
-	 cr2image.cpp          \
-	 crwimage.cpp          \
-	 datasets.cpp          \
-	 easyaccess.cpp        \
-	 epsimage.cpp          \
-	 error.cpp             \
-	 exif.cpp              \
-	 futils.cpp            \
-	 fujimn.cpp            \
-	 gifimage.cpp          \
-	 http.cpp              \
-	 image.cpp             \
-	 ini.cpp               \
-	 iptc.cpp              \
-	 jp2image.cpp          \
-	 jpgimage.cpp          \
-	 makernote.cpp         \
-	 metadatum.cpp         \
-	 minoltamn.cpp         \
-	 mrwimage.cpp          \
-	 nikonmn.cpp           \
-	 olympusmn.cpp         \
-	 orfimage.cpp          \
-	 panasonicmn.cpp       \
-	 pentaxmn.cpp          \
-	 pgfimage.cpp
+CCSRC =  basicio.cpp           \
+         bigtiffimage.cpp      \
+         bmpimage.cpp          \
+         canonmn_int.cpp       \
+         casiomn_int.cpp       \
+         convert.cpp           \
+         cr2image_int.cpp      \
+         cr2image.cpp          \
+         crwimage_int.cpp      \
+         crwimage.cpp          \
+         datasets.cpp          \
+         easyaccess.cpp        \
+         epsimage.cpp          \
+         error.cpp             \
+         exif.cpp              \
+         fujimn_int.cpp        \
+         futils.cpp            \
+         gifimage.cpp          \
+         http.cpp              \
+         image.cpp             \
+         ini.cpp               \
+         iptc.cpp              \
+         jp2image.cpp          \
+         jpgimage.cpp          \
+         makernote_int.cpp     \
+         metadatum.cpp         \
+         minoltamn_int.cpp     \
+         mrwimage.cpp          \
+         nikonmn_int.cpp       \
+         olympusmn_int.cpp     \
+         orfimage_int.cpp      \
+         orfimage.cpp          \
+         pgfimage.cpp
 ifdef HAVE_LIBZ
-CCSRC += pngimage.cpp      \
-	 pngchunk.cpp
+CCSRC += pngimage.cpp          \
+         pngchunk_int.cpp
 endif
-CCSRC += preview.cpp       \
-	 properties.cpp        \
-	 psdimage.cpp          \
-	 rafimage.cpp          \
-	 rw2image.cpp          \
-	 samsungmn.cpp
+CCSRC += preview.cpp           \
```

Comment:
I am confused about this. Lines 105-113 and 117-130 seems to be files that are always required. I would move those lines to the initial initialisation of `CCSRC` (After line 100). In that way we clearly indicate which are the files that are always required and at the end we add files conditionally (HAVE_LIBZ, USE_SSH, ENABLE_VIDEO, etc)

Reply:
I agree.  It would be better to declare all the "basic" files.  Then add HAVE_FOO when required.

I suspect it's been like this forever and you're the first to say "Why is it so convoluted?".

If I update the code on my machine, do I only have to stage & commit the correction.

Reply:
Yes that would be enough. You could also squash your 2 commits into just 1, and do `git push -f` to overwrite the commit that it is already in the remote. 

If you do not feel comfortable with that, we can take care of squashing the commits once everything is green here (just before merging the branch)